vital/tests/test_mesh_io.cxx

```diff
@@ -0,0 +1,160 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief test core mesh_io functionality
+
+#include <kwiversys/SystemTools.hxx>
+#include <tests/test_gtest.h>
+#include <tests/test_scene.h>
+#include <vital/exceptions.h>
+
+#include <vital/io/mesh_io.cxx>
+
+#include <gtest/gtest.h>
+
+kwiver::vital::path_t g_data_dir;
+
+// ----------------------------------------------------------------------------
+int main(int argc, char** argv)
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  GET_ARG(1, g_data_dir);
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class mesh_io : public ::testing::Test
+{
+  TEST_ARG(data_dir);
+
+  void compare_meshes(const kwiver::vital::mesh_sptr& first,
+                      const kwiver::vital::mesh_sptr& second)
+  {
```

Comment:
I'm not totally sure if this is how I should be comparing meshes to check the data is the same. I believe this is exhaustive enough (at least for the data that gets set in the cube_mesh function, I'm not checking edges or normals here), but I'm not sure if there's a better way to do this